pkg/tsdb/cloudwatch/query_transformer.go

```diff
@@ -104,10 +99,119 @@ func (e *cloudWatchExecutor) transformQueryResponsesToQueryResult(cloudwatchResp
 			queryResult.ErrorString = "Cloudwatch GetMetricData error: Too many datapoints requested - your search has been limited. Please try to reduce the time range"
 		}
 
+		eq, err := json.Marshal(executedQueries)
+		if err != nil {
+			plog.Error("Could not marshal executedString struct", err)
+		}
+
+		link, err := buildDeepLink(refID, requestQueries, executedQueries, startTime, endTime)
+		if err != nil {
+			plog.Error("Could not build deep link", err)
+		}
+
+		for _, frame := range frames {
+			frame.Meta = &data.FrameMeta{
+				ExecutedQueryString: string(eq),
+			}
+
+			if link != "" {
+				for _, field := range frame.Fields {
+					field.Config = &data.FieldConfig{
+						Links: []data.DataLink{
+							{
+								Title:       "View in CloudWatch console",
+								TargetBlank: true,
+								URL:         link,
+							},
+						},
+					}
+				}
+			}
+		}
+
 		queryResult.Dataframes = tsdb.NewDecodedDataFrames(frames)
-		queryResult.Meta.Set("gmdMeta", queryMeta)
 		results[refID] = queryResult
 	}
 
 	return results
 }
+
+// Generates a deep link from Grafana to the CloudWatch console. The link params are based on metric(s) for a given query row in the Query Editor
```

Comment:
Even though this isn't exported, it's best to follow the standard doc comment style.
```suggestion
// buildDeepLink generates a deep link from Grafana to the CloudWatch console. The link params are based on metric(s) for a given query row in the Query Editor.
```